process/block/alteredaccounts/alteredAccountsProvider_test.go

```diff
@@ -417,7 +538,124 @@ func testExtractAlteredAccountsFromPoolAddressHasBalanceChangeEsdtAndfNft(t *tes
 
 	encodedAddr := hex.EncodeToString([]byte("addr"))
-	require.Equal(t, 2, len(res[encodedAddr].Tokens))
+	require.Len(t, res[encodedAddr].Tokens, 2)
+}
+
+func testExtractAlteredAccountsFromPoolAddressHasMultipleNfts(t *testing.T) {
+	t.Parallel()
+
+	expectedToken0 := esdt.ESDigitalToken{
+		Value: big.NewInt(37),
+	}
+	expectedToken1 := esdt.ESDigitalToken{
+		Value: big.NewInt(38),
+		TokenMetaData: &esdt.MetaData{
+			Nonce: 5,
+			Name:  []byte("nft-0"),
+		},
+	}
+	expectedToken2 := esdt.ESDigitalToken{
+		Value: big.NewInt(37),
+		TokenMetaData: &esdt.MetaData{
+			Nonce: 6,
+			Name:  []byte("nft-1"),
+		},
+	}
+	args := getMockArgs()
+	args.AccountsDB = &state.AccountsStub{
+		LoadAccountCalled: func(_ []byte) (vmcommon.AccountHandler, error) {
+			return &state.UserAccountStub{
+				RetrieveValueFromDataTrieTrackerCalled: func(key []byte) ([]byte, error) {
+					if strings.Contains(string(key), "esdttoken") {
+						tokenBytes, _ := args.Marshalizer.Marshal(expectedToken0)
+						return tokenBytes, nil
+					}
+					if strings.Contains(string(key), "nft-0") {
+						tokenBytes, _ := args.Marshalizer.Marshal(expectedToken1)
+						return tokenBytes, nil
+					}
+					if strings.Contains(string(key), "nft-1") {
+						tokenBytes, _ := args.Marshalizer.Marshal(expectedToken2)
+						return tokenBytes, nil
+					}
+
+					return nil, nil
+				},
+			}, nil
+		},
+	}
+	aap, _ := NewAlteredAccountsProvider(args)
+
+	res, err := aap.ExtractAlteredAccountsFromPool(&indexer.Pool{
+		Txs: map[string]data.TransactionHandler{
+			"hash0": &transaction.Transaction{
+				SndAddr: []byte("addr"),
+			},
+		},
+		Logs: []*data.LogData{
+			{
+				LogHandler: &transaction.Log{
+					Address: []byte("addr"),
+					Events: []*transaction.Event{
+						{
+							Address:    []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTTransfer),
+							Topics: [][]byte{
+								[]byte("esdttoken"),
+							},
+						},
+						{
+							Address:    []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTNFTTransfer),
+							Topics: [][]byte{
+								expectedToken1.TokenMetaData.Name,
+								big.NewInt(0).SetUint64(expectedToken1.TokenMetaData.Nonce).Bytes(),
+							},
+						},
+						{
+							Address:    []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTNFTTransfer),
+							Topics: [][]byte{
+								expectedToken2.TokenMetaData.Name,
+								big.NewInt(0).SetUint64(expectedToken2.TokenMetaData.Nonce).Bytes(),
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	encodedAddr := hex.EncodeToString([]byte("addr"))
+	require.Len(t, res[encodedAddr].Tokens, 3)
+
+	for _, token := range res[encodedAddr].Tokens {
+		if token.Identifier == "esdttoken" {
+			require.Equal(t, &indexer.AccountTokenData{
+				Identifier: "esdttoken",
+				Balance:    expectedToken0.Value.String(),
+				Nonce:      0,
+				MetaData:   nil,
+			}, token)
+		}
+		if token.Identifier == "nft-o" {
```

Comment:
This test is truly _"interesting"_  .
You have no token identifier == `nft-o` -> note there is an `o`, not a `0`, **yet** your test passes => **WRONG**.
That is because your code never reaches this `if`, therefore never calls `require.Equal`.
Also, this `for loop` is also not required(useless), since len(ret)==3 AND you have 3 `ifs`.
Perhaps use 3 require.Contains conditions for each token and drop the for loop

Reply:
I still think this `for loop` could be replaced with a simple `require.Contains`. This way, if you want to add another entry in logs for this test, you don't have to also change the counter, the `require.Contains` will take care of it

Reply:
Done. However, I need that for loop to iterate over the slice and extract the tokens, but then I check all the extracted tokens, so if any of them is missing, the test will fail